xwiki-pro-macros-test/xwiki-pro-macros-test-docker/src/test/it/com/xwiki/pro/test/ui/GenericMacrosIT.java

```diff
@@ -76,61 +101,582 @@
+        assertEquals("NestedPanelsTestTitle2", panel3.getTitleText());
+        assertNull(panel3.getTitleBackgroundColor());
+        assertNull(panel3.getTitleColor());
+
+        // Content section 3rd panel.
+        assertEquals("Content4 for PanelMacroTest", panel3.getContentText());
+        assertNull(panel3.getContentBackgroundColor());
+        assertNull(panel3.getContentColor());
+    }
+
+    @Test
+    @Order(8)
+    void microsoftStreamMacroTest(TestUtils setup, TestReference testReference)
+    {
+        setup.createPage(testReference, createContent("microsoftstream-macros.vm"), "MicrosoftStreamTest");
+
+        MicrosoftStreamMacroPage page = new MicrosoftStreamMacroPage();
+
+        MicrosoftStreamMacro mStream1 = page.getMacro(0);
+        MicrosoftStreamMacro mStream2 = page.getMacro(1);
+        MicrosoftStreamMacro mStream3 = page.getMacro(2);
+
+        // There should be 3 MicrosoftStream macros.
+        assertEquals(3, page.getMacroCount());
+
+        // Checks the alignment of the macro.
+        assertEquals("right", mStream1.getAlignment());
+
+        // Checks the width of the macro, default = "500px".
+        assertEquals("600px", mStream1.getWidth());
+
+        // Checks the height of the macro, default = "300px".
+        assertEquals("400px", mStream1.getHeight());
+
+        // Checks the autoplay property, default = "false".
+        assertTrue(mStream1.hasAutoplay());
+        // Checks the showInfo property, default = "false".
+        assertTrue(mStream1.hasShowInfo());
+        // Checks the StartTime property existence.
+        assertTrue(mStream1.hasStartTime());
+        // Checks the StartTime property.
+        assertEquals(convertStartTime("01:12:13"), mStream1.getStartTime());
+        // Checks the actual URL of the microsoftStream.
+        assertTrue(mStream1.hasCorrectURL("www.youtube.com"));
+
+        // Checks the 2nd MicrosoftStream macro, with left alignment, personalized dimensions and set StartTime.
+        assertEquals("left", mStream2.getAlignment());
+        assertEquals("500px", mStream2.getWidth());
+        assertEquals("300px", mStream2.getHeight());
+        assertFalse(mStream2.hasAutoplay());
+        assertFalse(mStream2.hasShowInfo());
+        assertTrue(mStream2.hasStartTime());
+        assertEquals(convertStartTime("03:12:13"), mStream2.getStartTime());
+        assertTrue(mStream2.hasCorrectURL("www.youtube.com"));
+
+        // Checks the 2nd MicrosoftStream macro, with center alignment
+        assertEquals("center", mStream3.getAlignment());
+        assertEquals("500px", mStream3.getWidth());
+        assertEquals("300px", mStream3.getHeight());
+        assertFalse(mStream3.hasAutoplay());
+        assertFalse(mStream3.hasShowInfo());
+        assertFalse(mStream3.hasStartTime());
+        assertTrue(mStream3.hasCorrectURL("www.youtube.com"));
+    }
+
+    // Function to convert color codes from hex to rgb.
+    private String hexToRgb(String hex)
```

Comment:
Could you check if there is already some existing method you could use https://stackoverflow.com/questions/4129666/how-to-convert-hex-to-rgb-using-java ?